ndcube/utils/wcs.py

```diff
@@ -197,18 +201,31 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    extra_coords_list = [] # initiating new list
+    pix_coords = [0] * len(item_) # Set up a list of pixel coords as input to all_pix2world
```

Comment:
So I added an initiation for `pix_coords` as per request before the start of the for loop around `item_`.

Reply:
Perhaps this line should be moved into the ```for``` loop.  It's a tiny bit less efficient, but it resets the pixel coordinates each time so the behaviour is more predictable.

Reply:
Yup, done, moved initiation statement for pix_coords to inside for loop.